### DISCUSSION TITLE
How to show squiggles under the warrings, infos,

### DISCUSSION THREAD

**Discussion by jianghuachen28** [James-Yu/LaTeX-Workshop#4141] [Q&A] [answered]
Hi, 

There used to be squiggles under the warrings and infos, but they are missing these day and I don't find the resason. I changed the setting jason file to show the error background color, but the info background and foreground color are not working in my case. I used different color for warnings and infos but still not working. 

I would like to have the a wavy underline of pictures "Overfull" info. Would anyone provide any solutions? Many thanks.

Kasen

`
// English grammer check
"ltex.enabled": true, 
"ltex.language": "en-US",
"workbench.colorTheme": "Monokai High Contrast",

// 
"latex-workshop.message.error.show": false,
"latex-workshop.message.warning.show": false,
"latex-workshop.showContextMenu": true,
"latex-workshop.intellisense.update.aggressive.enabled": true,
"diffEditor.wordWrap": "off",
"editor.fontSize": 19,
"security.workspace.trust.untrustedFiles": "open",
"editor.wordWrap": "on",
"update.mode": "none",
"extensions.autoUpdate": "onlySelectedExtensions",
"update.showReleaseNotes": false,
"update.enableWindowsBackgroundUpdates": false,
"javascript.updateImportsOnFileMove.enabled": "never",
"typescript.updateImportsOnFileMove.enabled": "never",
"extensions.autoCheckUpdates": false,
"markdown.preview.scrollPreviewWithEditor": false,
"markdown.preview.scrollEditorWithPreview": false,
"editor.accessibilitySupport": "off",

// highlight
"workbench.colorCustomizations": {
   "editor.selectionBackground": "#f0f70e" ,
    "editor.selectionHighlightBackground": "#d9c843", 
    "editor.lineHighlightBackground": "#0cdeed3d",
    "editor.lineHighlightBorder": "#fe000030",
//    "editor.wordHighlightBackground": "#ff0000",
//    "editor.wordHighlightBorder":  "#ff0000",
   "editor.worldHighlightStrongBackground": "#ff0000",
//    "editor.wordHighlightStrongBorder": "#ff0000",

    "editorError.background": "#8c932b",
    "editorError.foreground": "#ff0000",
    "editorWarning.background": "#9d9b9b",
    "editorWarning.foreground": "#ff0000",
    // "editorInfo.background": "#8c932b",
    "editorInfo.foreground": "#ff0000",
    
    "list.errorForeground": "#ff0000",
    // "minimap.warningHighlight": "#ff0000"
    // "editorOverviewRuler.warningForeground": "#ff0000"
    // "statusBarItem.warningBackground": "#ff0000"
    // "editor.inlineValuesBackground": "#ff0000"
// "editorMarkerNavigation.background": "#ff0000"
// "testing.message.info.decorationForeground": "#ff0000",
// "testing.message.error.lineBackground": "#ff0000"
// "descriptionForeground": "#ff0000"

},
"window.zoomLevel": 1,
"editor.scrollPredominantAxis": false,
"editor.quickSuggestions": {
    "comments": "inline",
    "strings": "inline"
},
"editor.quickSuggestionsDelay": 5,
"terminal.integrated.shellIntegration.suggestEnabled": true,
"editor.inlineSuggest.suppressSuggestions": true,
"editor.suggest.shareSuggestSelections": true,
"git.openRepositoryInParentFolders": "never",
"emmet.showSuggestionsAsSnippets": true,
"editor.snippetSuggestions": "top",
`

**Comment by jlelong** [comment]
Nothing has changed in LW recently regarding warnings detection. The rendering of warning/infos is handled by VS Code itself.

**Reply by jianghuachen28** [reply]
Thanks for your reply!

Would you please provide some tip on how to handle this in VS code setting? I do not know how to show the squiggles  under warnings. infors, correctly. 

Only the Error settings are working,
`"editorError.background": "#8c932b",
"editorError.foreground": "#ff0000",`

Warning and Info are not working,
`"editorWarning.background": "#9d9b9b",
"editorWarning.foreground": "#ff0000",
// "editorInfo.background": "#8c932b",
"editorInfo.foreground": "#ff0000",`

Tried different methods, but failed.


Thanks for you time on my questions.

Kasen

**Reply by jlelong** [reply]
That's strange as setting
```json
    "workbench.colorCustomizations": {
        "editorWarning.background": "#ff0000"
    }
```
adds a red background to all warnings.

**Comment by jianghuachen28** [comment]
Hi，thank you for the reply.

I just found that the squiggles under infos are shown in tables that also overfull, but not shown in figure that overfull. Do you have any idea for this? Why is the overfull info not shown in the .tex catlog. Picture overfull info is only shown under pdf.catlog. See the following pictures.


![A](https://github.com/James-Yu/LaTeX-Workshop/assets/20107167/d3722242-4053-45b1-954e-0bbb9cd53c69)

![B](https://github.com/James-Yu/LaTeX-Workshop/assets/20107167/38856d13-6efc-4127-91e2-e4cafb800d56)

**Comment by jlelong** [comment]
The screenshots are not crystal clear which file reports which warnings. But if I understand well, the overfull in figure is not listed under the tex file. If so, there may be something wrong in parsing LaTeX compiler's log. Could you open provide a minimal example to reproduce the issue along with the extension and the compiler logs?

**Comment by jianghuachen28** [comment]
Thank you for the advice. I deleted all the other information in tex except the figure. The squiggles is working properly. It must be something wrong in the text. Dont know the reason. If time consuming, just ignore this inconvenience.


![image](https://github.com/James-Yu/LaTeX-Workshop/assets/20107167/d81edac2-249b-489d-b184-84e73b5e382f)

**Reply by jianghuachen28** [reply, marked as answer]
As discussed above, it is a PDF problem. I made a diagram in PPT and converted to PDF via "Save as Adobe PDF", and encounter the mentioned problem. 

When ticked the "Create PDF/A-1a:2005 compliant file" at the option, the squiggles will show up correctly.

![A](https://github.com/James-Yu/LaTeX-Workshop/assets/20107167/2aee93d3-7be6-4082-b875-8a702f882bdb)
